deploy/runtools/run_farm.py

```diff
@@ -111,30 +114,96 @@ class RunFarm(metaclass=abc.ABCMeta):
 
     Attributes:
         args: Set of options from the 'args' section of the YAML associated with the run farm.
+        default_simulation_dir: default location of the simulation dir on the run farm host
+        SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS: dict of host handles to number of FPGAs available
+        SIM_HOST_HANDLE_TO_MAX_METASIM_SLOTS: dict of host handles to number of metasim slots available
+        SIM_HOST_HANDLE_TO_SWITCH_ONLY_OK: dict of host handles to whether an instance is allowed to be used to hold only a switch simulation and nothing else
+
+        SORTED_SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS: sorted 'SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS' by FPGAs available
+        SORTED_SIM_HOST_HANDLE_TO_MAX_METASIM_SLOTS: sorted 'SIM_HOST_HANDLE_TO_MAX_METASIM_SLOTS' by metasim slots available
+
+        run_farm_hosts_dict: list of instances requested (Inst object and one of [None, boto3 object, mock boto3 object, other cloud-specific obj]). TODO: improve this later
+        mapper_consumed: dict of allocated instance names to number of allocations of that instance name.
+            this mapping API tracks instances allocated not sim slots (it is possible to allocate an instance
+            that has some sim slots unassigned)
+
     """
 
-    def __init__(self, args: Dict[str, Any]) -> None:
+    SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS: Dict[str, int]
+    SIM_HOST_HANDLE_TO_MAX_METASIM_SLOTS: Dict[str, int]
+    SIM_HOST_HANDLE_TO_SWITCH_ONLY_OK: Dict[str, bool]
+
+    SORTED_SIM_HOST_HANDLE_TO_MAX_FPGA_SLOTS: List[Tuple[int, str]]
+    SORTED_SIM_HOST_HANDLE_TO_MAX_METASIM_SLOTS: List[Tuple[int, str]]
+
+    run_farm_hosts_dict: Dict[str, List[Tuple[Inst, Optional[Union[EC2InstanceResource, MockBoto3Instance]]]]]
+    mapper_consumed: Dict[str, int]
+
+    default_simulation_dir: str
+
+    def __init__(self, args: Dict[str, Any], metasimulation_enabled: bool) -> None:
         self.args = args
+        self.metasimulation_enabled = metasimulation_enabled
```

Comment:
Missing class comment and type of variable (i.e. `metasim_enabled: bool` 2-3 lines above)

Reply:
fixed